### PR TITLE
get rid of more deprecation warnings in `OnlineDB`

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

Title says it all, getting rid of one more (hopefully last) deprecation warning in the `OnlineDB` area, thanks to the new flagging from core s/w group (see https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cfc78f/23964/build-logs/)
Technical, part of the migrations in https://github.com/cms-sw/cmssw/issues/36404.
Profited to clean-up headers, unused methods and merging plugin header with implementation.

#### PR validation:

`cmssw` compiles

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

File: OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc [hunk 1]
```diff
@@ -1,15 +1,47 @@
...
+
+/**
+   @class testSiStripConfigDb
+   @author R.Bainbridge
+   @brief Simple class that tests SiStripConfigDb service
+*/
+class testSiStripConfigDb : public edm::one::EDAnalyzer<> {
+public:
+  testSiStripConfigDb(const edm::ParameterSet&);
+  ~testSiStripConfigDb();
```
Comment (tvami) [inline OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc:28]: should this be moved to the default destructor?

Comment (mmusich) [reply OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc:28]: no the destructor is not trivial:
https://github.com/cms-sw/cmssw/blob/9bd0dbd0474b5546afa1b8e4d3240989663d7135/OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc#L42-L45

File: OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc [hunk 2]
```diff
@@ -1,15 +1,47 @@
...
+
+  void beginJob();
+  //void analyze(const edm::Event&, const edm::EventSetup&) = default;
```
[the remark this answers is not in the document]
Comment (mmusich) [reply OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc:31]: whoops, will remove.

[the remark this answers is not in the document]
Comment (mmusich) [reply OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc]: done in the last push.

File: OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc (hunk 1 above)
Comment (tvami) [reply OnlineDB/SiStripConfigDb/test/plugins/testSiStripConfigDb.cc:28]: ok thanks!

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by mmusich** [comment]
@cmsbuild, please test
   * third time is a charm

**Comment by tvami** [comment]
+db

**Comment by perrotta** [comment]
+1

**Pull request merged**